tmk_core/common/action_util.c

```diff
@@ -51,8 +51,18 @@ extern inline void clear_keys(void);
 static int8_t oneshot_mods = 0;
 static int8_t oneshot_locked_mods = 0;
 int8_t get_oneshot_locked_mods(void) { return oneshot_locked_mods; }
-void set_oneshot_locked_mods(int8_t mods) { oneshot_locked_mods = mods; }
-void clear_oneshot_locked_mods(void) { oneshot_locked_mods = 0; }
+void set_oneshot_locked_mods(int8_t mods) {
+    if (mods != oneshot_locked_mods) {
+        oneshot_locked_mods = mods;
+        oneshot_locked_mods_set_kb(mods);
+    }
+}
+void clear_oneshot_locked_mods(int8_t mods) {
```

Comment:
If this has been changed, then this needs to be changed as well: 
https://github.com/qmk/qmk_firmware/blob/f4012ba9b9abc357847cc4a3d0e696b933687a85/tmk_core/common/action_util.h#L76

Reply:
Also, these probably should be `uint8_t`

```suggestion
void clear_oneshot_locked_mods(uint8_tmods) {
```